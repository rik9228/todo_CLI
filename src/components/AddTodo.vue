<template>
  <div class="addForm">
    <p>Todoを追加する</p>
    <!-- value は自己のコンポーネントデータを制御するから別にv-modelで良い -->
    <input class="addForm__form" type="text" v-model="value">
    <!-- 親にイベントを投げる -->
    <button class="addForm__button" @click="onSubmit">追加</button>
  </div>
</template>

<script>
export default {
  data: function () {
    return {
      id: 0,
      value: '',
      state: false
    }
  },
  methods: {
    onSubmit: function () {
      if (this.value === '') {
        alert('項目名を入力してください')
        return
      }
      this.$emit('addTodo', this.id, this.value, this.state)
      this.id++
      this.value = ''
      this.state = false
    }
  }
}
</script>

<style scoped>

.addForm{
  margin-top: 16px;
}

.addForm__form{
  padding: 0.3em;
}

.addForm__button{
  margin-left: 16px;
}

.addForm__button{
  border: 0;
  background-color: #00df38;
  color:#fff;
  padding: 0.3em 1.5em;
}
</style>
