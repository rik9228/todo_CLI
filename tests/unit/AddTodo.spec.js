import { shallowMount } from '@vue/test-utils'
import AddTodo from '@/components/AddTodo.vue'

describe('AddTodo.vue', () => {
  it('snapshot', () => {
    const wrapper = shallowMount(AddTodo)
    expect(wrapper).toMatchSnapshot()
  })
  it('入力できる', async () => {
    const wrapper = shallowMount(AddTodo)
    const input = wrapper.find('input')
    // フォームのvalueを変更
    input.element.value = 'hogehoge'
    // フォームの入力イベントを発火
    await input.trigger('input')
    // v-modelでdataのvalueが'hogehoge'に変わっているかどうか
    expect(wrapper.vm.$data.value).toBe('hogehoge')
  })
  it('追加できる', async () => {
    const wrapper = shallowMount(AddTodo)
    const button = wrapper.find('button')
    const input = wrapper.find('input')
    input.element.value = 'hogehoge'
    await input.trigger('input')
    expect(wrapper.vm.$data.value).toBe('hogehoge')
    expect(wrapper.vm.$data.id).toBe(0)
    expect(wrapper.vm.$data.state).toBe(false)
    // emitされた
    expect(wrapper.emitted().addTodo).toBeUndefined()
    // クリックイベントを実行する
    await button.trigger('click')
    // emitされて送られた値を確認
    expect(wrapper.emitted().addTodo[0]).toEqual([0, 'hogehoge', false])
    expect(wrapper.vm.$data.value).toBe('')
    expect(wrapper.vm.$data.id).toBe(1)
    expect(wrapper.vm.$data.state).toBe(false)
  })
})
