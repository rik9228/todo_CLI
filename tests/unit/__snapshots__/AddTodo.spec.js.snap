// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo.vue snapshot 1`] = `<div class="addForm"><input type="text" class="addForm__form"> <button class="addForm__button">追加</button></div>`;
